CODING_STANDARDS.md

```diff
@@ -0,0 +1,378 @@
+# JavaScript CodeStyle
+
+In an effort to have consistent READABLE code across our various JavaScript libraries these coding rules should be followed. This will make it easier for clients/new hires/other developers to read any of our code as it will be written in a single format
+
+---
+
+## General Provisions
+
+Follow the linting (eslint/prettier) rules setup for each library which should mostly be the same and auto-format on file save or pre-commit hook
+
+---
+
+## Additional guidelines that the linters will not cover 
+
+### __Naming__
+
+Self-documenting code is the ideal we want to have so semantic naming should be as descriptive as necessary and consistent. The only exception is the indexing variable in an loop. That can be shortened to a single letter starting from `i`. For an absolute global `const` such as milliseconds in a day use all-caps with words separated by an underscore `_`.
+
+- variableNamesLikeThis
+- methodNamesLikeThis
+- ClassNamesLikeThis
+- ComponentsLikeThis
+- `const DAY_IN_MILLISECONDS = 86400000;`
+
+### __Spacing__
+
+Let the code breathe instead of cramming it into as tiny of a space as possible as readability is more important. For example: if there is a file with numerous hooks being used in a row and then a final return, let them all be spaced out and own an individual piece of real estate on your screen instead of being shoved all together. Keep your covid distance or give your dog a little yard to run around in as 🏠-🏠-🏠-🏠-🏠 is better than 🏠🏠🏠🏠🏠
+
+### __Exports__
+
+Use one or multiple `export const` or `export type` and never `export default`
+
+### __Imports__
```

Comment:
I would rather let the TSServer organize the imports automatically - I think it does it already.

https://eshlox.net/2019/12/02/vscode-automatically-organize-typescript-imports

Reply:
> I would rather let the TSServer organize the imports automatically - I think it does it already.
> 
> https://eshlox.net/2019/12/02/vscode-automatically-organize-typescript-imports

+10 on this one!!

Reply:
Petyo modified an eslint rule to allow for this to be possible without this organize imports thing which we'll use instead